qiita_db/support_files/patches/python_patches/23.py

```diff
@@ -0,0 +1,195 @@
+ALTER TABLE qiita.{0} ADD COLUMN sample_type varchar;
+ALTER TABLE qiita.{0} ADD COLUMN collection_timestamp timestamp;
+ALTER TABLE qiita.{0} ADD COLUMN host_subject_id varchar;
+ALTER TABLE qiita.{0} ADD COLUMN description varchar;
+ALTER TABLE qiita.{0} ADD COLUMN latitude float8;
+ALTER TABLE qiita.{0} ADD COLUMN longitude float8;
+ALTER TABLE qiita.{0} ADD COLUMN required_sample_info_status varchar;"""
+
+sql_update = """UPDATE qiita.{0}
+                SET physical_specimen_location=%s,
+                    physical_specimen_remaining=%s,
+                    dna_extracted=%s,
+                    sample_type=%s,
+                    collection_timestamp=%s,
+                    host_subject_id=%s,
+                    description=%s,
+                    latitude=%s,
+                    longitude=%s,
+                    required_sample_info_status=%s
+                WHERE sample_id=%s"""
+
+for s_id in study_ids:
+    # Get the data from the requried_sample_info
+    data = conn_handler.execute_fetchall(sql_select, (s_id,))
+
+    # Add rows to the study_sample_columns table
+    # 10 -> the number of rows that we are adding, see `sql_cols`
+    conn_handler.add_to_queue(queue_name, sql_cols, [s_id] * 10)
+
+    # Get the dynamic table name
+    table_name = "sample_%d" % s_id
+
+    # Add the columns to the dynamic table
+    conn_handler.add_to_queue(queue_name, sql_alter.format(table_name))
+
+    # Add the values for each sample
+    for sample_data in data:
+        sample_data = dict(sample_data)
+        conn_handler.add_to_queue(
+            queue_name, sql_update.format(table_name),
+            (sample_data['physical_location'],
+             sample_data['has_physical_specimen'],
+             sample_data['has_extracted_data'],
+             sample_data['sample_type'],
+             sample_data['collection_timestamp'],
+             sample_data['host_subject_id'],
+             sample_data['description'],
+             sample_data['latitude'],
+             sample_data['longitude'],
+             sample_data['required_sample_info_status'],
+             sample_data['sample_id']))
+
+# Once all data has been moved, we can drop the columns from the
+# requried_sample_info table; and drop the required_sample_info_status table
+sql = """ALTER TABLE qiita.required_sample_info
+            DROP COLUMN physical_location,
+            DROP COLUMN has_physical_specimen,
+            DROP COLUMN has_extracted_data,
+            DROP COLUMN sample_type,
+            DROP COLUMN required_sample_info_status_id,
+            DROP COLUMN collection_timestamp,
+            DROP COLUMN host_subject_id,
+            DROP COLUMN description,
+            DROP COLUMN latitude,
+            DROP COLUMN longitude"""
+conn_handler.add_to_queue(queue_name, sql)
+
+# Since that table no longer stores required metadata,
+# we are going to rename it
+sql = """ALTER TABLE qiita.required_sample_info
+            RENAME TO study_sample"""
```

Comment:
I will move this to SQL but just FYI, since you are changing the name of the table you don't need the schema. Actually, if you put the schema does not work (I've tested it on my system cause I also thought it was needed)